pkg/apis/provisioning/v1alpha5/kubelet_configuration.go

```diff
@@ -22,4 +22,10 @@ type KubeletConfiguration struct {
 	// Note that not all providers may use all addresses.
 	//+optional
 	ClusterDNS []string `json:"clusterDNS,omitempty"`
+	// MaxPods is the maximum number of Pods that can run on the Kubelet. While the value must be
```

Comment:
Thoughts on defaulting this to 110?

Reply:
Seems like there's a tradeoff for small nodes, since the overhead will be so large. You have to reserve headroom for 110 even if you'll only ever use 1 pod.

Another approach would be to use some function e.g. Max(5%, 110pods). I'm wary of exposing too low level control over max pods, since the kubelet flag assumptions were not built with karpenter in mind. Ideally karpenter would build a simple strategy(s) that fits all use cases.

Reply:
I thought about it, but that is a change from how things are working right now where the default is whatever the EKS optimized image is doing (defaulting to the VPC CNI settings). So, we'd want to have a way to know whether we use that or not. I figured having a default would mean we would need yet another setting to have it fallback to the ENI count approach currently used. That make sense?

Reply:
My idea was that `nil` meant "do nothing", i.e use whatever EKS AMI does, which at the moment is assuming non-prefix VPC CNI. EKS Managed node groups using prefix VPC CNI set 110 as default on all instance types with < 30 cores, and a rather curious default of 250 for > 30 cores (lots of very, very small pods on the same node). In other words, outside of a "legacy" VPC CNI configuration, kubelet almost always runs with this set to 110. I don't see why this would be a problem for Karpenter.